src/gl_fns.rs

```diff
@@ -293,6 +293,24 @@ impl Gl for GlFns {
         }
     }
 
+    // https://www.khronos.org/registry/OpenGL-Refpages/es2.0/xhtml/glGetUniform.xml
+    fn get_uniform_iv(&self, program: GLuint, location: GLint) -> Vec<GLint> {
+        let mut result: Vec<GLint> = Vec::new();
+        unsafe {
+            self.ffi_gl_.GetUniformiv(program, location, result.as_mut_ptr());
```

Comment:
I am suspicious about this code, since the vector's length will not be affected by the implementation of GetUniformiv. How many elements does this AP return?

Reply:
According to spec, it depends on the `location` argument and it can be between 1 and 16 elements. One way to fix this would be to preallocate it with length 16, but probably to `match location` and determine the length is better. What do you think?

Reply:
It would be safest to have an array of 16 elements on the stack which is modified by GetUniformiv, then match on location to obtain a slice of the stack array that can be turned into a Vec to return.

Reply:
Done.
Just curious, why would it be safer in comparison to
```Rust
let len = get_uniform_vector_length(&location);
let mut result: Vec<_> = Vec::with_capacity(len);
unsafe {
    self.ffi_gl_.GetUniformiv(program, location, result.as_mut_ptr();
}
```
?

Reply:
If there were a bug in get_uniform_vector_length where it returned a number less than the actual number of elements written, GetUniformiv could write past the end of the allocated space. Additionally, when starting with a vector with a certain capacity, we still need to manually set the length of the vector to reflect the number of elements written. If we start with an array with a fixed size which GetUniformiv should never exceed, we can build a vector from the results safely even if get_uniform_vector_length contains a bug that takes the wrong number of elements from the slice.

Reply:
I see. That's great to know, thanks!